src/sentry/dynamic_sampling/latest_release_booster.py

```diff
@@ -20,6 +23,67 @@ class TooManyBoostedReleasesException(Exception):
     pass
 
 
+@dataclass(frozen=True)
+class BoostedRelease:
+    id: int
+    timestamp: float
+    environment: Optional[str]
+
+    def to_augmented(self, release: Release, project_id: int) -> "ExtendedBoostedRelease":
+        return ExtendedBoostedRelease(
+            id=self.id,
+            timestamp=self.timestamp,
+            environment=self.environment,
+            version=release.version,
+            platform=Platform(self._get_project_platform_from_release(release, project_id)),
+        )
+
+    @staticmethod
+    def _get_project_platform_from_release(release: Release, project_id: int) -> Optional[str]:
+        try:
+            return release.projects.get(id=project_id).platform  # type:ignore
+        except Project.DoesNotExist:
+            # If we don't find the project of this release we just default to having no platform name in the
+            # BoostedRelease.
+            return None
+
+
+@dataclass(frozen=True)
+class ExtendedBoostedRelease(BoostedRelease):
+    version: str
+    platform: Platform
+
+
+@dataclass
+class BoostedReleases:
+    boosted_releases: List[BoostedRelease] = field(default_factory=list)
+
+    def add_release(self, id: int, timestamp: float, environment: Optional[str]) -> None:
+        self.boosted_releases.append(
+            BoostedRelease(id=id, timestamp=timestamp, environment=environment)
+        )
+
+    def to_extended_boosted_releases(
+        self, project_id: int, limit: int
+    ) -> List[ExtendedBoostedRelease]:
+        # We get release models in order to have all the information to extend the releases we get from the cache.
+        models = self._get_releases_models(limit)
+        return [
+            boosted_release.to_augmented(release=models[boosted_release.id], project_id=project_id)
+            for boosted_release in self.boosted_releases
+            if boosted_release.id in models
+        ]
+
+    def _get_last_release_ids(self, last: int) -> List[int]:
+        return [boosted_release.id for boosted_release in self.boosted_releases[-last:]]
```

Comment:
```suggestion
        return [boosted_release.id for boosted_release in self.boosted_releases[-limit:]]
```
do you mean `limit`?